src/vim.js

```diff
@@ -1148,19 +1143,59 @@ export function initVim(CodeMirror) {
       }
     }
 
+    var specialKey = {
+      Return: 'CR', Backspace: 'BS', 'Delete': 'Del', Escape: 'Esc', Insert: 'Ins',
+      ArrowLeft: 'Left', ArrowRight: 'Right', ArrowUp: 'Up', ArrowDown: 'Down',
+      Enter: 'CR', ' ': 'Space'
+    };
+    var ignoredKeys = { Shift: 1, Alt: 1, Command: 1, Control: 1,
+      CapsLock: 1, AltGraph: 1, Dead: 1, Unidentified: 1 };
```

Comment:
maybe we should move these somewhere shared? looks like ignoredKeys is still duplicated in cm_adapter

Reply:
i've managed to remove this duplication by  moving cm5 style keymapping to cm5 file.